plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -76,6 +77,114 @@ public void testMigrateTable(IcebergFileFormat fileFormat)
+                "ARRAY[4, 5], " +
+                "CAST(map(ARRAY['key2'], ARRAY['value2']) AS map(varchar, varchar)), " +
+                "CAST(row(2) AS row(d integer)))";
+        assertUpdate("INSERT INTO " + icebergTableName + secondRow, 1);
+        assertQuery("SELECT count(*) FROM " + icebergTableName, "VALUES 2");
+        assertThat(query("SELECT * FROM " + icebergTableName))
+                .matches(firstRow + " UNION ALL " + secondRow);
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMigrateTableParquetSchemaEvolution()
+            throws Exception
+    {
+        String randomNameSuffix = randomNameSuffix();
+        String tableNameOneColumn = "test_migrate_one_column_" + randomNameSuffix;
+        String tableNameTwoColumns = "test_migrate_two_columns_" + randomNameSuffix;
+        String hiveTableNameOneColumn = "hive.tpch." + tableNameOneColumn;
+        String hiveTableNameTwoColumns = "hive.tpch." + tableNameTwoColumns;
+        String icebergTableNameTwoColumns = "iceberg.tpch." + tableNameTwoColumns;
+
+        assertUpdate("CREATE TABLE " + hiveTableNameOneColumn + " WITH (format = 'PARQUET') AS SELECT 1 col1", 1);
+        assertUpdate("CREATE TABLE " + hiveTableNameTwoColumns + " WITH (format = 'PARQUET') AS SELECT 2 col1, CAST(row(10, 20) AS row(x integer, y integer)) AS nested", 1);
+
+        // Copy the parquet file containing only one column to the table with two columns
+        Path tableNameOneColumnLocation = Path.of("%s/tpch/%s".formatted(dataDirectory, tableNameOneColumn));
+        Path tableNameTwoColumnsLocation = Path.of("%s/tpch/%s".formatted(dataDirectory, tableNameTwoColumns));
+        try (Stream<Path> files = Files.list(tableNameOneColumnLocation)) {
+            Path file = files.filter(path -> !path.getFileName().toString().startsWith(".")).collect(onlyElement());
+            Files.copy(file, tableNameTwoColumnsLocation.resolve(file.getFileName()));
+        }
+
+        assertUpdate("CALL iceberg.system.migrate('tpch', '" + tableNameTwoColumns + "')");
+
+        assertThat(query("SELECT * FROM " + icebergTableNameTwoColumns))
+                .skippingTypesCheck()
+                .matches("VALUES (1, CAST(null AS row(x integer, y integer))), (2, row(10, 20))");
+
+        assertUpdate("INSERT INTO " + icebergTableNameTwoColumns + " VALUES (3, row(100, 200))", 1);
+        assertThat(query("SELECT * FROM " + icebergTableNameTwoColumns))
+                .skippingTypesCheck()
+                .matches("VALUES (1, CAST(null AS row(x integer, y integer))), (2, row(10, 20)), (3, row(100, 200))");
+
+        assertUpdate("DROP TABLE " + icebergTableNameTwoColumns);
+    }
+
+    @Test
+    public void testMigrateTableParquetRowColumnSchemaEvolution()
+            throws Exception
+    {
+        String randomNameSuffix = randomNameSuffix();
+        String tableNameRowOneColumn = "test_migrate_row_one_column_" + randomNameSuffix;
```

Comment:
Sorry, I've made a little mistake yesterday

`Column` should be renamed to `Field`

https://trino.io/docs/current/language/types.html#row

The row has fields, not columns.